dockerfile-generator/dockerfile-gen-services/src/test/java/io/hyscale/dockerfile/gen/services/manager/impl/DockerScriptManagerTest.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.dockerfile.gen.services.manager.impl;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.util.CollectionUtils;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.models.SupportingFile;
+import io.hyscale.dockerfile.gen.services.model.DockerfileGenContext;
+import io.hyscale.dockerfile.gen.services.util.ServiceSpecTestUtil;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+
+@SpringBootTest
+public class DockerScriptManagerTest {
+
+    @Autowired
+    private DockerScriptManagerImpl dockerScriptManagerImpl;
+
+    public static Stream<Arguments> input() {
+        return Stream.of(Arguments.of("/input/command-script/script-and-cmd.hspec"),
+                Arguments.of("/input/command-script/config-script.hspec"),
+                Arguments.of("/input/command-script/run-script.hspec"),
+                Arguments.of("/input/command-script/run-cmd.hspec"),
+                Arguments.of("/input/command-script/config-cmd.hspec"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("input")
+    public void scriptManagerTest(String serviceSpecPath) throws IOException {
```

Comment:
Aren't both artifactManagerTest and scriptManagertest same ?

Reply:
Artifact Manager test specifically tests that relative location for artifacts is updated in supporting files.
Whereas Script Manager Test if for script and command-based supporting files, both test different implementation of DockerfileEntityManager